roxie/ccd/ccdfile.cpp

```diff
@@ -1710,10 +1710,12 @@ class CRoxieFileCache : implements IRoxieFileCache, implements ICopyFileProgress
                 dlfn.clearForeign();
 #ifdef _CONTAINERIZED
             const char *defaultDir = defaultPlaneDirPrefix;
+            bool defaultDirPerPart = defaultPlaneDirPerPart;
 #else
             const char *defaultDir = nullptr;
+            bool defaultDirPerPart = false;
 #endif
-            makePhysicalPartName(dlfn.get(), partNo, numParts, localLocation, replicationLevel, DFD_OSdefault, defaultDir);
+            makePhysicalPartName(dlfn.get(), partNo, numParts, localLocation, replicationLevel, DFD_OSdefault, defaultDir, defaultDirPerPart);
```

Comment:
Doesn't this need to pass in a flag to indicate if there is a subdir per part for this file?  I think it would only be a problem if copying from a bare metal system.

Reply:
Discussed with Richard, but I still not sure.

This is building the localLocation to copy to, or if useRemoteResources and copyResources are false (which I believe is the default in cloud), then no copy is involved.

But doesn't that mean that if the default roxie data plane is different (different prefix) from the plane of file it's looking up here, then the 'defaultDir' is also wrong here?

If so, shouldn't the directory used here to make the localLocation also be using the prefix from the file?
In which case, it wasn't going to work before this change if the planes differed, and yes I think the subdirperpart flag should also come from the file.